private/bufpkg/bufimage/bufimagemodify/bufimagemodify.go

```diff
@@ -219,3 +247,27 @@ func int32SliceIsEqual(x []int32, y []int32) bool {
 	}
 	return true
 }
+
+func stringOverridesToBoolOverrides(stringOverrides map[string]string) (map[string]bool, error) {
+	validatedOverrides := map[string]bool{}
+	for fileImportPath, overrideString := range stringOverrides {
+		overrideBool, err := strconv.ParseBool(overrideString)
+		if err != nil {
+			return nil, fmt.Errorf("non-boolean override %s set for file %s", overrideString, fileImportPath)
+		}
+		validatedOverrides[fileImportPath] = overrideBool
+	}
+	return validatedOverrides, nil
+}
+
+func overrideStringToFileOptionOptimizeMode(overrideString string) (descriptorpb.FileOptions_OptimizeMode, error) {
+	switch overrideString {
+	case "SPEED":
+		return descriptorpb.FileOptions_SPEED, nil
+	case "CODE_SIZE":
+		return descriptorpb.FileOptions_CODE_SIZE, nil
+	case "LITE_RUNTIME":
+		return descriptorpb.FileOptions_LITE_RUNTIME, nil
+	}
+	return descriptorpb.FileOptions_SPEED, fmt.Errorf("invalid optimize for value: %s", overrideString)
```

Comment:
We should be able to use the generated enum association for this. It's a neat trick for a quick validation, and it prevents us from having to update this code when/if any other values are added to `descriptorpb`.
```suggestion
	optimizeMode, ok := descriptorpb.FileOptions_OptimizeMode_value[overrideString]
	if !ok {
		return descriptorpb.FileOptions_SPEED, fmt.Errorf("invalid optimize for value: %s", overrideString)
	}
	return descriptorpb.FileOptions_OptimizeMode(optimizeMode), nil
```

Reply:
Nice! I was trying to figure out how to do that, haha :D Thanks!